docs/api_proposal.md

```diff
@@ -0,0 +1,54 @@
+# Taxonomy API proposal
+
+DINA taxonomy API methods. Expanded from [Taxonomy API discussion](https://docs.google.com/spreadsheets/d/1uaJh1qt6mvY0ZCEB6uBKdnUnyGdpQnZcnFqTMHTS5pA/edit#gid=0) google doc.
+
+One goal is to harmonize with existing taxonomy APIs, which benefits downstream users. Makes it more likely that our services can be incorporated into integration projects such as [taxize](https://cran.r-project.org/web/packages/taxize/index.html) R package. 
+
+See [DINA glossary](https://github.com/DINA-Web/dina-use-cases/blob/master/glossary.md) for terms. 
+
+## Assumptions
+
+* there will be some canonical root URL, format to be determined
+* taxonomic names are not guaranteed to be unique, therefore cannot safely perform CRUD options based on name strings without disambiguation 
+* the API will be versioned, and older versions may be deprecated in the future
+
+## Methods
+
+* /taxon : return info about a taxon
+* /search : return potentially matching taxa
+* /mrca : return the most recent common ancestor of two or more taxa
+* /about : return information about the taxonomy
+* /rank : return the rank of a taxon
+* /subtree : return the subtree descended from a taxon
+
+## Questions for discussion
+
+* how much taxonomy editing functionality will we provide through the API?
+** delete a taxon (re-assigning children to parent?)
+** move a taxon (and children)
+** merge taxa
+** rename taxon
+** split taxon
+
+* limits / pagination
+** on methods like subtree, limit levels / number of children returned
+** lineage / children
+
+* IDs vs names
```

Comment:
The IDs that are exposed in the restfu apis are internal ids, i.e. db primary keys. 64bit integers

All other ids are simply that, other ids that are stored in addition to the primary keys. These can be accessible with a filter, like:
`/taxon?filter[doi]=...`

`/taxon?filter[gbifid]=...`

This would include the ID that AAFC assigns to the entity in question.